### PR TITLE
Updated graph teams tag sample to use delegated permissions

### PR DESCRIPTION
**Pull request by Prithvi-MSFT**
Changed the sample to use delegated permissions instead of application permissions

### CODE REVIEW

File: samples/graph-teams-tag/csharp/GraphTeamsTag/Controllers/TeamTagController.cs [hunk 1]
```diff
@@ -6,23 +6,56 @@ namespace GraphTeamsTag.Controllers
...
         private readonly ILogger<TeamTagController> _logger;
 
         private readonly GraphHelper graphHelper;
 
-        public TeamTagController(ILogger<TeamTagController> logger, GraphHelper graphHelper)
+        private readonly IConfiguration _configuration;
+
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        private readonly IHttpContextAccessor _httpContextAccessor;
```
Comment (ChetanSharma-msft) [inline samples/graph-teams-tag/csharp/GraphTeamsTag/Controllers/TeamTagController.cs:27]: Add summary for these.

File: samples/graph-teams-tag/csharp/GraphTeamsTag/Helper/GraphHelper.cs [hunk 2]
```diff
@@ -93,18 +100,26 @@ public async Task<IEnumerable<TeamTag>> ListTeamworkTagsAsync(string teamId)
...
-                .Request()
-                .GetAsync();
+            try
+            {
+                var teamworkTag = await this.graphBetaClient.Teams[teamId].Tags[teamTagId]
+               .Request()
+               .GetAsync();
 
-            var teamwTagDto = new TeamTag
+                var teamwTagDto = new TeamTag
```
Comment (ChetanSharma-msft) [inline samples/graph-teams-tag/csharp/GraphTeamsTag/Helper/GraphHelper.cs:109]: Rename it as "teamTag"

File: samples/graph-teams-tag/csharp/GraphTeamsTag/Provider/SimpleBetaGraphClient.cs [hunk 3]
```diff
@@ -14,11 +14,17 @@ public class SimpleBetaGraphClient
...
         /// </summary>
-        private readonly IOptions<AzureSettings> azureSettings;
+        private readonly IConfiguration _configuration;
 
-        public SimpleBetaGraphClient(IOptions<AzureSettings> azureSettings)
+        private static readonly string ClientIdConfigurationSettingsKey = "AzureAd:ClientId";
+
+        private static readonly string ClientSecretConfigurationSettingsKey = "AzureAd:AppSecret";
+
+        private static readonly string TenantIdConfigurationSettingsKey = "AzureAd:TenantId";
```
Comment (ChetanSharma-msft) [inline samples/graph-teams-tag/csharp/GraphTeamsTag/Provider/SimpleBetaGraphClient.cs:23]: Add summary for these.

File: samples/graph-teams-tag/csharp/GraphTeamsTag/Program.cs [hunk 4]
```diff
@@ -18,9 +21,9 @@
 builder.Services.AddOptions<AzureSettings>()
 .Configure<IConfiguration>((botOptions, configuration) =>
 {
```
Comment (ChetanSharma-msft) [inline samples/graph-teams-tag/csharp/GraphTeamsTag/Program.cs:23]: Please push latest appsettings.json

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my review comments.

File: samples/graph-teams-tag/csharp/GraphTeamsTag/Provider/SimpleBetaGraphClient.cs (hunk 3 above)
Comment (Prithvi-MSFT) [reply samples/graph-teams-tag/csharp/GraphTeamsTag/Provider/SimpleBetaGraphClient.cs:27]: Added

File: samples/graph-teams-tag/csharp/GraphTeamsTag/Controllers/TeamTagController.cs (hunk 1 above)
Comment (Prithvi-MSFT) [reply samples/graph-teams-tag/csharp/GraphTeamsTag/Controllers/TeamTagController.cs:37]: Added

File: samples/graph-teams-tag/csharp/GraphTeamsTag/Program.cs (hunk 4 above)
Comment (Prithvi-MSFT) [reply samples/graph-teams-tag/csharp/GraphTeamsTag/Program.cs:21]: Added latest app settings

**Review by ChetanSharma-msft — APPROVED** [review]
Changes looks fine, approving the PR.

**Review by Wajeed-msft — APPROVED** [review]

**Comment by ChetanSharma-msft** [comment]
**Executed build pipeline:** https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/3613115369
**Result:**

**Comment by Prithvi-MSFT** [comment]
Build pipeline status - All builds passed
image
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/3617265065

**Pull request merged**